### ISSUE TITLE
ICRA paper

### ISSUE THREAD

**Issue by pagand** [pagand/ORL_optimizer#24]
1- Applying 3 constraints to both train and evaluation (suggestion, create a utility func for them and call in train and val)
1.1. model sensitivity >> purturb input and measure variance of output, if larger than thershold, only remove that transition (step)
1.2. VAE >> reward penalization 
1.3. OOD >> if Elbo loss is too large for a transition (step), truncate the trajectory, what ever the value of done was

2- training in parallel
Curriculum learning idea
2.1. unpack  D4RL to List[List] trajectory
2.2. sort the list key equal to length of them
2.3. choose random len(traje)/ batchsize  from the list and select batch size after it (consider edge case)
2.4. select a window of size seq_len + future_len (10+3) 
2.5. predict all future_len at once and compute error with weight decay
2.6. apply the back propagation
2.7. slide the window one and replace the last element with the predicted
2.8. continue until the len of the first element of the batch allows
2.9. do for more epochs (50)

3.
